src/dune_rules/modules_field_evaluator.mli

```diff
@@ -19,10 +19,16 @@ type kind =
   | Implementation of Implementation.t
   | Exe_or_normal_lib
 
+type melange_conf =
```

Comment:
You're taking this record as an argument but then you're immediately flattening it. Why not just take the flat list of arguments at this point? If you want to keep the record, I think it makes more sense to define it as `Dune_file.Buildable.Modules.t` and then make it a part of `Buildable.t` and `Melange_emit`.

Reply:
> You're taking this record as an argument but then you're immediately flattening it. Why not just take the flat list of arguments at this point?

This allowed to leave a single function while applying minimal changes the buildable call points (just wrap them with `Buildable` variant).

> I think it makes more sense to define it as Dune_file.Buildable.Modules.t and then make it a part of Buildable.t and Melange_emit.

`Buildable.t` is used in so many places. I don't mind going down that path, but I'd rather do it in a separate branch as I am not sure how extensive the changes can be. I am also not sure what would o into this new `Buildable.Modules.t` type: just `modules_field` and `modules_without_implementation`? Something else?

I removed `melange_conf` type in https://github.com/ocaml/dune/pull/6349/commits/73bab6beb25bfc75db0b094e9adc85b6c6d996a0, there are a few arguments now but I think it's the "less bad" solution (although I have no strong opinion).